__tests__/functional/update-script.js

```diff
@@ -0,0 +1,456 @@
+
+    child_process.execFileSync('git', ['add', '.'], { cwd: src })
+    child_process.execFileSync('git', ['commit', '-m', 'Initial commit'], { cwd: src })
+
+    // let's imagine our prototype has been customised slightly
+    let config = fs.readFileSync(path.join(src, 'app', 'config.js'), 'utf8')
+    config = config.replace('Service name goes here', 'Jest test service')
+    fs.writeFileSync(path.join(src, 'app', 'config.js'), config, 'utf8')
+
+    // and it has been run
+    fs.writeFileSync(path.join(src, 'usage-data-config.json'), '{}')
+
+    child_process.execFileSync('git', ['commit', '-m', 'Test', '-a'], { cwd: src })
+
+    // populate the update folder to speed up tests
+    child_process.execSync(`unzip -q ${fixtureDir}/${headReleaseArchiveFilename}`, { cwd: src })
+    child_process.execSync(`mv ${headReleaseBasename} update`, { cwd: src })
+    fs.copyFileSync(path.join(fixtureDir, headReleaseArchiveFilename), path.join(src, 'update', headReleaseArchiveFilename))
+  }
+
+  function mktestPrototypeSync (dest) {
+    const src = path.resolve(fixtureDir, 'prototype')
+    try {
+      fs.accessSync(src)
+    } catch (error) {
+      _mktestPrototypeSync(src)
+    }
+
+    if (dest) {
+      child_process.execSync(`cp -r ${src} ${dest}`)
+      return dest
+    } else {
+      return src
+    }
+  }
+
+  beforeAll(() => {
+    process.chdir(tmpDir)
+    console.log('Running tests in temporary directory', process.cwd())
+
+    // setup fixtures - running this now saves time later
+    fs.mkdirSync(fixtureDir)
+    mktestArchiveSync()
+    mktestPrototypeSync()
+  })
+
+  /*
+   * Tests
+   */
+
+  describe('check', () => {
+    it('exits with error if run in an empty folder', () => {
+      const testDir = 'empty'
+      fs.mkdirSync(testDir)
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it('exits with error if folder does not contain a package.json file', () => {
+      const testDir = 'no-package-json'
+      fs.mkdirSync(testDir)
+      fs.writeFileSync(path.join(testDir, 'foo'), 'my important data about govuk-prototype-kit')
+      fs.writeFileSync(path.join(testDir, 'bar'), "don't delete my data!")
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it('exits with error if package.json file does not contain string govuk-prototype-kit', () => {
+      const testDir = mktestDirSync('no-govuk-prototype-kit')
+      fs.writeFileSync(path.join(testDir, 'package.json'), '{\n  "name": "test"\n}')
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it('exits with error if package.json file contains string containing govuk-prototype-kit', () => {
+      const testDir = mktestDirSync('name-contains-govuk-prototype-kit')
+      fs.writeFileSync(path.join(testDir, 'package.json'), '{\n  "name": "govuk-prototype-kit-test"\n}')
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it.skip('exits with error if folder does not contain a VERSION.txt file', () => {
```

Comment:
FYI there's a test here that's being skipped

Reply:
Ah yeah, this is because I think it would be useful functionality but I haven't got around to implementing it yet...